### PR TITLE
Fix some export issues. Remove EXPORTABLE_CODE as export parameter.

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
1. Fix exportable code issue CVS-142373, checked that CVS-142263 isn't reproduced, added changes according to CVS-135469

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — DISMISSED** [review]

File: src/otx/core/model/base.py [hunk 1]
```diff
@@ -965,6 +967,39 @@ def check_if_quantized(model: openvino.Model) -> bool:
...
+        to_exportable_code: bool = False,
+    ) -> Path:
+        """Export this model to the specified output directory.
+
+        Args:
+            output_dir (Path): directory for saving the exported model
+            base_name: (str): base name for the exported model file. Extension is defined by the target export format
+            export_format (OTXExportFormatType): format of the output model
+            precision (OTXExportPrecisionType): precision of the output model
+            to_exportable_code (OTXExportPrecisionType): whether to generate exportable code
```
Comment (eunwoosh) [inline src/otx/core/model/base.py]:
```suggestion
            to_exportable_code (bool, optional): whether to generate exportable code
```

File: src/otx/core/model/base.py [hunk 2]
```diff
@@ -622,6 +622,7 @@ def export(
         base_name: str,
         export_format: OTXExportFormatType,
         precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        to_exportable_code: bool = False,
```
Comment (eunwoosh) [inline src/otx/core/model/base.py:625]: Could you add a docstring for this argument?

File: src/otx/engine/engine.py [hunk 3]
```diff
@@ -490,6 +490,7 @@ def export(
         export_format: OTXExportFormatType = OTXExportFormatType.OPENVINO,
         export_precision: OTXPrecisionType = OTXPrecisionType.FP32,
         explain: bool = False,
+        export_demo_package: bool = False,
```
Comment (eunwoosh) [inline src/otx/engine/engine.py:493]: Could you add a docstring for this?

File: src/otx/core/exporter/base.py [hunk 4]
```diff
@@ -82,6 +82,7 @@ def export(
         base_model_name: str = "exported_model",
         export_format: OTXExportFormatType = OTXExportFormatType.OPENVINO,
         precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        to_exportable_code: bool = False,
```
Comment (eunwoosh) [inline src/otx/core/exporter/base.py:85]: Could you add docstring for this?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work :) I left some comments. Please take a look.

**Review by harimkang — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Pull request merged**